library/string.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package library
+
+import (
+	"encoding/base64"
+	"strconv"
+	"strings"
+
+	"github.com/buildkite/yaml"
+	json "github.com/ghodss/yaml"
+)
+
+// ToString is a helper function to convert
+// the provided interface value to a string.
+func ToString(v interface{}) string {
+	switch v := v.(type) {
+	case string:
+		return v
+	case bool:
+		return strconv.FormatBool(v)
+	case []byte:
+		return base64.StdEncoding.EncodeToString(v)
+	case float32:
+		return strconv.FormatFloat(float64(v), 'g', -1, 32)
+	case float64:
+		return strconv.FormatFloat(v, 'g', -1, 64)
+	case int:
+		return strconv.Itoa(v)
+	case int8:
+		return strconv.FormatInt(int64(v), 10)
+	case int16:
+		return strconv.FormatInt(int64(v), 10)
+	case int32:
+		return strconv.FormatInt(int64(v), 10)
+	case int64:
+		return strconv.FormatInt(v, 10)
+	case uint:
+		return strconv.FormatUint(uint64(v), 10)
+	case uint8:
+		return strconv.FormatUint(uint64(v), 10)
+	case uint16:
+		return strconv.FormatUint(uint64(v), 10)
+	case uint32:
+		return strconv.FormatUint(uint64(v), 10)
+	case uint64:
+		return strconv.FormatUint(v, 10)
+	case []interface{}:
+		return unmarshalSlice(v)
+	default:
+		return unmarshalMap(v)
+	}
+}
+
+// helper function to unmarshal a parameter in map format.
+func unmarshalMap(v interface{}) string {
+	yml, _ := yaml.Marshal(v)
+	out, _ := json.YAMLToJSON(yml)
+
+	return string(out)
+}
+
+// helper function to unmarshal a parameter in slice format.
+func unmarshalSlice(v interface{}) string {
+	out, _ := yaml.Marshal(v)
```

Comment:
Is there value in checking this error?

Reply:
I don't think so 👍 The amount of cases where this will actually error out are relatively low.

This function is copied directly from the [go-vela/compiler](https://github.com/go-vela/compiler/blob/master/compiler/native/unmarshal.go#L65-L79) repo and we've been utilizing it for quite some time with no issues 🙂